Userland/Libraries/LibVideo/Color/CodingIndependentCodePoints.h

```diff
@@ -0,0 +1,222 @@
+    ICtCp = 14,
+};
+
+enum class ColorRange : u8 {
+    Studio = 0, // 16-2
+    Full = 1,
+};
+
+// https://en.wikipedia.org/wiki/Coding-independent_code_points
+struct CodingIndependentCodePoints {
+public:
+    constexpr CodingIndependentCodePoints(ColorPrimaries color_primaries, TransferCharacteristics transfer_characteristics, MatrixCoefficients matrix_coefficients, ColorRange color_range)
+        : m_color_primaries(color_primaries)
+        , m_transfer_characteristics(transfer_characteristics)
+        , m_matrix_coefficients(matrix_coefficients)
+        , m_color_range(color_range)
+    {
+    }
+
+    inline ColorPrimaries color_primaries() const { return m_color_primaries; }
+    inline void set_color_primaries(ColorPrimaries value) { m_color_primaries = value; }
+    inline TransferCharacteristics transfer_characteristics() const { return m_transfer_characteristics; }
+    inline void set_transfer_characteristics(TransferCharacteristics value) { m_transfer_characteristics = value; }
+    inline MatrixCoefficients matrix_coefficients() const { return m_matrix_coefficients; }
+    inline void set_matrix_coefficients(MatrixCoefficients value) { m_matrix_coefficients = value; }
+    inline ColorRange color_range() const { return m_color_range; }
+    inline void set_color_range(ColorRange value) { m_color_range = value; }
+
+    inline void default_code_points_if_unspecified(ColorPrimaries cp, TransferCharacteristics tc, MatrixCoefficients mc)
+    {
+        if (color_primaries() == ColorPrimaries::Unspecified)
+            set_color_primaries(cp);
+        if (transfer_characteristics() == TransferCharacteristics::Unspecified)
+            set_transfer_characteristics(tc);
+        if (matrix_coefficients() == MatrixCoefficients::Unspecified)
+            set_matrix_coefficients(mc);
+    }
+
+private:
+    ColorPrimaries m_color_primaries;
+    TransferCharacteristics m_transfer_characteristics;
+    MatrixCoefficients m_matrix_coefficients;
+    ColorRange m_color_range;
+};
+
+constexpr StringView color_primaries_to_string(ColorPrimaries color_primaries)
+{
+    switch (color_primaries) {
+    case ColorPrimaries::BT709:
+        return "BT.709"sv;
+    case ColorPrimaries::Unspecified:
+        return "Unspecified"sv;
+    case ColorPrimaries::BT470M:
+        return "BT.470M"sv;
+    case ColorPrimaries::BT470BG:
+        return "BT.470 B/G"sv;
+    case ColorPrimaries::BT601:
+        return "BT.601"sv;
+    case ColorPrimaries::SMPTE240:
+        return "SMPTE 240M"sv;
+    case ColorPrimaries::GenericFilm:
+        return "Generic Film"sv;
+    case ColorPrimaries::BT2020:
+        return "BT.2020"sv;
+    case ColorPrimaries::XYZ:
+        return "XYZ"sv;
+    case ColorPrimaries::SMPTE431:
+        return "SMPTE 431"sv;
+    case ColorPrimaries::SMPTE432:
+        return "SMPTE 432"sv;
+    case ColorPrimaries::EBU3213:
+        return "EBU Tech 3213"sv;
+    default:
+        return "Reserved"sv;
```

Comment:
Yeah, I think you're right. All the other values are defined as Reserved as well, but I can work around this by specifically including Reserved = 0 for CP and TC switches and then returning `"Reserved"` outside the switch as well so that if new values are ever added, they must be included in the to_string functions.